PyInstaller/utils/win32/icon.py

```diff
@@ -224,6 +224,18 @@ def splitter(s):
     else:
         logger.info("Copying icons from %s", srcpath)
 
+    # Bail out quickly if the input is invalid. Letting images in the wrong
+    # format be passed to Window's API gives very cryptic error messages as
+    # it's generally unclear why PyInstaller would treat an image file as an
+    # executable.
+    if srcext != ".exe":
+        raise SystemExit(
```

Comment:
A `ValueError` would be more suited, maybe?

Reply:
I was mirroring the behavior of [L249](https://github.com/pyinstaller/pyinstaller/blob/8825887df336ae7ee8b56a25c87450b8333c0aa4/PyInstaller/utils/win32/icon.py#L249) which also calls `SystemExit`. But I would generally prefer it be a `ValueError`.

Reply:
Actually, sod consistency. I'm changing it.

Reply:
I see that function chooses to raise a `ValueError` if it gets more than one `.exe` so consistency is already ruined :stuck_out_tongue: ...

Reply:
Ah yes, L249 is brutal :)